generator/src/test/java/com/scottlogic/deg/generator/cucumber/steps/TypeRegistryConfiguration.java

```diff
@@ -37,7 +37,7 @@ public void configureTypeRegistry(TypeRegistry tr) {
 
         tr.defineParameterType(new ParameterType<>(
             "number",
-            "([+-]?\\d+(\\.\\d+)?)$",
+            "([+-]?\\d+(\\.\\d+)?)",
```

Comment:
This permits a `{number}` anywhere in the step definition, rather than requiring it to be at the end.